### PR TITLE
Skip building torchvision with ffmpeg when python==3.9

### PR DESCRIPTION
**Pull request by prabhat00155**
Building torchvision with ffmpeg 4.3 causes seg fault with python 3.9. Initial debugging shows crash in ffmpeg's sws_scale(). Until this issue is resolved, it's better to avoid building torchvision with ffmpeg for python 3.9.

cc @pmeier

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -351,7 +351,12 @@ def get_extensions():
 
     ffmpeg_exe = distutils.spawn.find_executable('ffmpeg')
     has_ffmpeg = ffmpeg_exe is not None
-    if sys.platform != 'linux':
+    # Building torchvision with ffmpeg on MacOS or with Python 3.9 causes
```
Comment (NicolasHug) [inline setup.py:354]: Should we add a FIXME or a TODO comment? It makes it easy to find out all outstanding issues with a simple `git grep command`

**Review by NicolasHug — APPROVED** [review]
Thank you @prabhat00155 , I think this is indeed the safest option for now. LGTM. Do you think we should reopen https://github.com/pytorch/vision/issues/3367?

File: setup.py [hunk 2]
```diff
@@ -351,7 +351,12 @@ def get_extensions():
...
+    # crash. See the following GitHub issues for more details.
+    # https://github.com/pytorch/pytorch/issues/65000
+    # https://github.com/pytorch/vision/issues/3367
+    if sys.platform != 'linux' or (
+            sys.version_info.major == 3 and sys.version_info.minor == 9):
```
Comment (NicolasHug) [inline setup.py:359]: BTW I think we can remove the ffmpeg version checks below as well, right?

File: setup.py (hunk 1 above)
Comment (prabhat00155) [reply setup.py]: Yeah, makes sense.

File: setup.py (hunk 2 above)
Comment (prabhat00155) [reply setup.py:359]: https://github.com/pytorch/vision/pull/4413 does that. After merging this PR, #4413 should be good to merge.

**Comment by fmassa** [comment]
One thing it would be good to understand is why only Python 3.9 is failing. IIUC the python version has no influence on the ffmpeg version we are installing, so maybe this is an issue with our binding code (torchbind?)

**Comment by prabhat00155** [comment]
> One thing it would be good to understand is why only Python 3.9 is failing. IIUC the python version has no influence on the ffmpeg version we are installing, so maybe this is an issue with our binding code (torchbind?)

Yeah, makes sense. I've created this issue: https://github.com/pytorch/vision/issues/4430 to track this, will debug this further.

**Pull request merged**